cmd/npmExecuteLint_test.go

```diff
@@ -0,0 +1,363 @@
+package cmd
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/mock"
+	"github.com/stretchr/testify/assert"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+)
+
+type mockLintUtilsBundle struct {
+	*mock.FilesMock
+	execRunner *mock.ExecMockRunner
+}
+
+func (u *mockLintUtilsBundle) getExecRunner() execRunner {
+	return u.execRunner
+}
+
+func (u *mockLintUtilsBundle) getGeneralPurposeConfig(configURL string) error {
+	u.AddFile(filepath.Join(".pipeline", ".eslintrc.json"), []byte(`abc`))
+	return nil
+}
+
+func newNpmMockUtilsBundle() mockLintUtilsBundle {
+	utils := mockLintUtilsBundle{FilesMock: &mock.FilesMock{}, execRunner: &mock.ExecMockRunner{}}
+	return utils
+}
+
+type npmExecuteOptions struct {
+	install            bool
+	runScripts         []string
+	runOptions         []string
+	defaultNpmRegistry string
+	sapNpmRegistry     string
+}
+
+type npmExecutorMock struct {
+	utils   mockLintUtilsBundle
+	options npmExecuteOptions
+}
+
+// FindPackageJSONFilesWithScript mocking implementation
+func (n *npmExecutorMock) FindPackageJSONFilesWithScript(packageJSONFiles []string, script string) ([]string, error) {
+	var packagesWithScript []string
+	for _, file := range packageJSONFiles {
+		fileContent, err := n.utils.FileRead(file)
+		if err != nil {
+			return nil, err
+		}
+		if strings.Contains(string(fileContent), script) {
+			packagesWithScript = append(packagesWithScript, file)
+		}
+	}
+	return packagesWithScript, nil
+}
+
+// ExecuteAllScripts mocking implementation
+func (n *npmExecutorMock) ExecuteAllScripts() error {
```

Comment:
Why do we have to mock this method with such a complex logic? I looks like you are re implementing the real ExecuteAllScripts. f we want to unit tests it should be okay to just check to the ExecuteAllScripts was called with the right parameters. Can we be sure that linting works together with the real ExecuteAllScripts? Why not using it here?